pkg/model/appconfig.go

```diff
@@ -78,11 +82,65 @@ func (b *AppConfig) addToModel(model *BackstageModel, _ bsv1.Backstage) (bool, e
 }
 
 // implementation of RuntimeObject interface
-func (b *AppConfig) updateAndValidate(m *BackstageModel, _ bsv1.Backstage) error {
+func (b *AppConfig) updateAndValidate(m *BackstageModel, backstage bsv1.Backstage) error {
+	err := b.updateDefaultBaseUrls(backstage)
+	if err != nil {
+		return err
+	}
 	updatePodWithAppConfig(m.backstageDeployment, m.backstageDeployment.container(), b.ConfigMap.Name, m.backstageDeployment.defaultMountPath(), "", true, maps.Keys(b.ConfigMap.Data))
 	return nil
 }
 
+// updateDefaultBaseUrls detects if it is running on OCP and tries to set the baseUrl in the default app-config,
+// per the cluster ingress domain and the Route spec.
+// Note that this is purposely done on a best effort basis. So it is not considered an issue if the cluster ingress domain
+// could not be determined, since the user can always set it explicitly in their custom app-config.
+func (b *AppConfig) updateDefaultBaseUrls(backstage bsv1.Backstage) error {
```

Comment:
I think this function has a lot of responsibilities.
Let's consider some separation of concerns by splitting it leaving app-config update part here and moving recognizing of route url to the route object. 
This way it is not needed to duplicate validations like isRouteEnabled and isOpenshift here because it is a prerogative of Route (which is Openshift object by definition).